modules/dr-recover-lost-control-plane-hosts.adoc

```diff
@@ -0,0 +1,305 @@
+// Module included in the following assemblies:
+//
+// * disaster_recovery/scenario-1-infra-recovery.adoc
+
+[id="dr-scenario-1-recover-master-hosts_{context}"]
+= Recovering from lost master hosts
+
+////
+TODO:
+ - Add a final result
+////
+
+Follow these steps to recover from the loss of one or more master hosts.
+
+.Prerequisites
+
+* Access to the cluster as a user with the `cluster-admin` role.
+* SSH access to a remaining master host.
+* A backed up etcd snapshot, if you are recovering a loss of a majority of masters.
+
+.Procedure
+
+. Restore etcd quorum on the remaining master.
++
+[NOTE]
+====
+This step is only necessary if you have had a majority of your masters fail. You can skip this step if you have a majority of your masters still available.
+====
+
+.. Copy the etcd snapshot file to the remaining master host.
++
+This procedure assumes that you have copied a snapshot file called `snapshot.db` to the `/home/core/` directory of your master host.
+
+.. Access the remaining master host.
+
+.. Set the `INITIAL_CLUSTER` variable to the list of members in the format of `<name>=<url>`. This variable will be passed to the restore script, and in this procedure, it is assumed that there is only a single member at this time.
++
+----
+[core@ip-10-0-143-125 ~]$ source /run/etcd/environment; export INITIAL_CLUSTER="etcd-member-$(hostname).$(hostname -d)=https://${ETCD_DNS_NAME}:2380"
+----
```

Comment:
@bergerhoffer We need to make this more generic in bare metal and vsphere we can not assume hostname will give us what is expected.

Let's change this to a more generic example

[core@ip-10-0-143-125 ~]$ export INITIAL_CLUSTER="etcd-member-ip-10-0-143-125=https://etcd-0.clustername.devcluster.openshift.com:2380"

Reply:
We could note the general format as
export INITIAL_CLUSTER="etcd-member-<hostname>.<domain>=https://<etcd-dns-name>:2380"